api/v0/proto/retrievability/types.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package celestia.v0.retrievability.types;
+
+// TODO: oneof different blob versions
+message Blob {
+  // The blob data.
+  bytes data = 1;
+  // The signer of the blob.
+  optional bytes signer = 2;
+  // The blob commitment.
+  bytes commitment = 3;
+}
+
+// Blob with blob proof. Either the blob and the proof are optional, depending
+// on the options of the request.
+message BlobWithProof {
+  // The blob.
+  optional Blob blob = 1;
+  // The proof of blob inclusion or non-inclusion.
+  optional DataRootProof proof = 2;
+  // Index of the first share of the blob in the ODS.
+  uint64 index = 3;
+}
```

Comment:
Do you need index here? The index of first share can be obtained from `Proof.Range.Start`

Reply:
Fixed in 7624dc2c226538fd09c8fd92b9a172eb2744e92d.